lib/tasks/prune_reports.rake

```diff
@@ -100,13 +100,25 @@ UNITS:
         puts "#{start_time.to_s(:db)}: Deleting #{deletion_count} orphaned records from #{model.table_name}"
         pbar = ProgressBar.new('Deleting', deletion_count, STDOUT)
 
-        # Deleting a very large group of records in MySQL can be very slow with no feedback
+        struct = database_settings
+        adapter = struct.adapter
+
+        # Deleting a very large group of records can be very slow with no feedback
         # Breaking the deletion up into blocks turns out to be overall faster
         # and allows for progress feedback
         DELETION_BATCH_SIZE = 1000
         while deletion_count > 0
+          case adapter
+            when 'mysql', 'mysql2', 'sqlite3'
+              command_line = "delete from #{model.table_name} where #{deletion_where_clause} limit #{DELETION_BATCH_SIZE}"
+            when 'postgresql'
+              command_line = "delete from #{model.table_name} where #{deletion_where_clause} DESC LIMIT #{DELETION_BATCH_SIZE}"
```

Comment:
it's correction at this bug:
Preparing to delete from resource_events
2015-09-14 16:10:19: Deleting 53021 orphaned records from resource_events
rake aborted!    0% |

      | ETA:  --:--:--
PG::SyntaxError: ERROR:  syntax error at or near "limit"
LINE 1: ...atus_id not in (select id from resource_statuses) limit 1000
                                                             ^
: delete from resource_events where resource_status_id not in (select id
from resource_statuses) limit 1000
/var/lib/gems/1.9.1/gems/activerecord-3.2.21/lib/active_record/connection_adapters/postgresql_adapter.rb:650:in
`async_exec'
/var/lib/gems/1.9.1/gems/activerecord-3.2.21/lib/active_record/connection_adapters/postgresql_adapter.rb:650:in
`block in execute'
/var/lib/gems/1.9.1/gems/activerecord-3.2.21/lib/active_record/connection_adapters/abstract_adapter.rb:280:in
`block in log'
/var/lib/gems/1.9.1/gems/activesupport-3.2.21/lib/active_support/notifications/instrumenter.rb:20:in
`instrument'
/var/lib/gems/1.9.1/gems/activerecord-3.2.21/lib/active_record/connection_adapters/abstract_adapter.rb:275:in
`log'
/var/lib/gems/1.9.1/gems/activerecord-3.2.21/lib/active_record/connection_adapters/postgresql_adapter.rb:649:in
`execute'
/usr/share/puppet-dashboard/lib/tasks/prune_reports.rake:108:in `block (4
levels) in <top (required)>'
/usr/share/puppet-dashboard/lib/tasks/prune_reports.rake:95:in `each'
/usr/share/puppet-dashboard/lib/tasks/prune_reports.rake:95:in `block (3
levels) in <top (required)>'
/var/lib/gems/1.9.1/gems/rake-0.9.6/lib/rake/task.rb:228:in `call'
/var/lib/gems/1.9.1/gems/rake-0.9.6/lib/rake/task.rb:228:in `block in
execute'
/var/lib/gems/1.9.1/gems/rake-0.9.6/lib/rake/task.rb:223:in `each'
/var/lib/gems/1.9.1/gems/rake-0.9.6/lib/rake/task.rb:223:in `execute'
/var/lib/gems/1.9.1/gems/rake-0.9.6/lib/rake/task.rb:166:in `block in
invoke_with_call_chain'
/usr/lib/ruby/1.9.1/monitor.rb:211:in `mon_synchronize'
/var/lib/gems/1.9.1/gems/rake-0.9.6/lib/rake/task.rb:159:in
`invoke_with_call_chain'
/var/lib/gems/1.9.1/gems/rake-0.9.6/lib/rake/task.rb:152:in `invoke'
/var/lib/gems/1.9.1/gems/rake-0.9.6/lib/rake/application.rb:143:in
`invoke_task'
/var/lib/gems/1.9.1/gems/rake-0.9.6/lib/rake/application.rb:101:in `block
(2 levels) in top_level'
/var/lib/gems/1.9.1/gems/rake-0.9.6/lib/rake/application.rb:101:in `each'
/var/lib/gems/1.9.1/gems/rake-0.9.6/lib/rake/application.rb:101:in `block
in top_level'
/var/lib/gems/1.9.1/gems/rake-0.9.6/lib/rake/application.rb:110:in
`run_with_threads'
/var/lib/gems/1.9.1/gems/rake-0.9.6/lib/rake/application.rb:95:in
`top_level'
/var/lib/gems/1.9.1/gems/rake-0.9.6/lib/rake/application.rb:73:in `block in
run'
/var/lib/gems/1.9.1/gems/rake-0.9.6/lib/rake/application.rb:160:in
`standard_exception_handling'
/var/lib/gems/1.9.1/gems/rake-0.9.6/lib/rake/application.rb:70:in `run'
/var/lib/gems/1.9.1/gems/rake-0.9.6/bin/rake:37:in `<top (required)>'
/usr/local/bin/rake:23:in `load'
/usr/local/bin/rake:23:in `<main>'
Tasks: TOP => reports:prune:orphaned

2015-09-16 10:04 GMT+02:00 Aaron Stone <notifications@github.com>:

> In lib/tasks/prune_reports.rake
> <https://github.com/sodabrew/puppet-dashboard/pull/333#discussion_r39602725>
> :
>
> >          # Breaking the deletion up into blocks turns out to be overall faster
> >          # and allows for progress feedback
> >          DELETION_BATCH_SIZE = 1000
> >          while deletion_count > 0
> > +          case adapter
> > +            when 'mysql', 'mysql2', 'sqlite3'
> > +              command_line = "delete from #{model.table_name} where #{deletion_where_clause} limit #{DELETION_BATCH_SIZE}"
> > +            when 'postgresql'
> > +              command_line = "delete from #{model.table_name} where #{deletion_where_clause} DESC LIMIT #{DELETION_BATCH_SIZE}"
>
> Does the DESC order of the where clause make a different to PostgreSQL
> correctness and/or performance?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/sodabrew/puppet-dashboard/pull/333/files#r39602725>.
>